src/validation.c

```diff
@@ -644,3 +649,31 @@ validate_default_route_consistency(__unused const NetplanParser* npp, GHashTable
     g_slist_free_full(defroutes, g_free);
     return ret;
 }
+
+gboolean
+validate_veth_pair(__unused const NetplanState* np_state, const NetplanNetDefinition* netdef, GError** error)
+{
+
+    NetplanNetDefinition* veth_peer = netdef->veth_peer_link;
+    if (veth_peer && veth_peer->type != NETPLAN_DEF_TYPE_NM_PLACEHOLDER_) {
```

Comment:
suggestion: Some brief comment/reference about the `NETPLAN_DEF_TYPE_NM_PLACEHOLDER_` quirk would probably be helpful in the future.